src/python/pants/backend/project_info/filter_targets.py

```diff
@@ -88,11 +104,21 @@ def filter_tag_regex(tag_regex: str) -> TargetFilter:
         regex = compile_regex(tag_regex)
         return lambda tgt: any(bool(regex.search(tag)) for tag in tgt.get(Tags).value or ())
 
+    def filter_granularity(granularity: TargetGranularity) -> TargetFilter:
+        if granularity == TargetGranularity.all_targets:
+            return lambda _: True
+        elif granularity == TargetGranularity.file_targets:
+            return lambda tgt: not tgt.address.is_base_target
+        else:
+            assert granularity == TargetGranularity.base_targets
+            return lambda tgt: tgt.address.is_base_target
```

Comment:
You can use `pants.util.enums.match`:

```python
return match({
  TargetGranularity.all_targets: lambda _: True,
  TargetGranularity.file_targets: lambda tgt: not tgt.address.is_base_target,
  TargetGranularity.base_targets: lambda tgt: tgt.address.is_base_target,
})
```

It'll validate that the match is exhaustive.